stl/inc/format

```diff
@@ -1743,6 +1743,65 @@ public:
 
 inline constexpr size_t _Fmt_buffer_size = 256;
 
+template <class _OutputIt, class _Ty>
+struct _Iterator_appender {
+    _OutputIt& _Out;
+
+    void _Append(_Ty* _First, _Ty* _Last) {
+        _Out = _STD _Copy_unchecked(_First, _Last, _STD move(_Out));
+    }
+};
+
+template <class _Container, class _Ty>
+concept _Appendable_container = requires(_Container& _Cont, _Ty* _First, _Ty* _Last) {
+    _Cont.append(_First, _Last);
+};
+
+template <class _Container>
+struct _Back_insert_iterator_container_access : back_insert_iterator<_Container> {
+    explicit _Back_insert_iterator_container_access(back_insert_iterator<_Container> _Iter)
+        : back_insert_iterator<_Container>(_Iter) {}
+
+    using back_insert_iterator<_Container>::container;
+};
+
+// clang-format off
+template <class _Container, class _Ty>
+    requires _Appendable_container<_Container, _Ty>
+struct _Iterator_appender<back_insert_iterator<_Container>, _Ty> {
+    // clang-format on
+private:
+    _Container& _Cont;
+
+public:
+    explicit _Iterator_appender(back_insert_iterator<_Container> _Out) noexcept
+        : _Cont(*_Back_insert_iterator_container_access<_Container>{_Out}.container) {} // TRANSITION, LLVM-41450
+
+    void _Append(_Ty* _First, _Ty* _Last) {
+        _Cont.append(_First, _Last);
+    }
+};
+
+// clang-format off
+template <class _Container, class _Ty>

```

Comment:
We could avoid some code duplication / simplify by merging these partial specializations of `_Iterator_appender` if you make the `requires-expression` on 1788 a named `_Insertable_container` concept a la:
```c++
template <class _Container, class _Ty>
concept _Insertable_container = requires(_Container& _Cont, _Ty* _First, _Ty* _Last) {
    _Cont.insert(_RANGES end(_Cont), _First, _Last);
};

template <class _Container, class _Ty>
    requires _Appendable_container<_Container, _Ty> || _Insertable_container<_Container, _Ty>
struct _Iterator_appender<back_insert_iterator<_Container, _Ty>> {
private:
    _Container& _Cont;

public:
    explicit _Iterator_appender(back_insert_iterator<_Container> _Out) noexcept
        : _Cont(*_Back_insert_iterator_container_access<_Container>{_Out}.container) {} // TRANSITION, LLVM-41450

    void _Append(_Ty* _First, _Ty* _Last) {
        if constexpr (_Appendable_container<_Container, _Ty>) {
            _Cont.append(_First, _Last);
        } else {
            _Cont.insert(_RANGES end(_Cont), _First, _Last);
        }
    }
};
```